### PR TITLE
## nmap -A baidu.com

### PR DESCRIPTION
**Pull request by chennqqi**
when run nmap with `-A` argument
 if not install nmap-scripts will cause error:
```
  NSE: failed to initialize the script engine:
  could not locate nse_main.lua
```

Signed-off-by: chennqqi <chennqqi@qq.com>

support map `-A` argument

## Description

without install nmap-script `nmap -A` will run failed

## Motivation and Context

support for full feature nmap function

- [x ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
- [ ] My issue has received approval from the maintainers or lead with the `design/approved` label

## How Has This Been Tested?

1. Deployed nmap via UI
2. Enter `-A www.baidu.com` or some other real domain/ip `-A 10.0.0.1`
 nmap will show message
image
image
3. enter into docker, then run `apk add --no-cache nmap-scripts`
4. redo step 1 you will see `nmap -A` run sucessfully

## Types of changes

- [x ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [ ] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: sample-functions/Nmap/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,13 @@
 FROM functions/alpine:latest
 
-RUN apk add --no-cache nmap
+## nmap -A baidu.com
```
Comment (burtonr) [inline sample-functions/Nmap/Dockerfile]: This comment block is unnecessary.

File: sample-functions/Nmap/Dockerfile [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
+##  NSE: failed to initialize the script engine:
+##  could not locate nse_main.lua
+##
 
-ENV fprocess="xargs nmap"
+RUN apk add --no-cache nmap nmap-scripts
 
-
-CMD ["fwatchdog"]
+ENV fprocess="xargs nmap"	ENV fprocess="xargs nmap"
```
Comment (burtonr) [inline sample-functions/Nmap/Dockerfile]: It looks like there was some issue that resulted in 2 `ENV` commands being here. 1 of them should be deleted

File: sample-functions/Nmap/Dockerfile [hunk 3]
```diff
@@ -1,8 +1,13 @@
...
+##
+##  NSE: failed to initialize the script engine:
+##  could not locate nse_main.lua
+##
 
-ENV fprocess="xargs nmap"
+RUN apk add --no-cache nmap nmap-scripts
 
-
-CMD ["fwatchdog"]
```
Comment (burtonr) [inline sample-functions/Nmap/Dockerfile]: This `CMD` command needs to be in place for the image to function properly

**Comment by saiyam1814** [comment]
@chennqqi  can you make the changes requested by @burtonr ?

**Comment by alexellis** [comment]
Closing as inactive. Thanks for the input from all.

**Pull request closed without merging**